engine/common/requester/engine.go

```diff
@@ -474,6 +471,9 @@ func (e *Engine) onEntityResponse(originID flow.Identifier, res *messages.Entity
 			Msg("onEntityResponse entries received")
 	}
 
+	e.unit.Lock()
```

Comment:
Move it down here, so that we don't lock the engine, when querying Identities with `e.state.Final().Identities`